tests/lib/rules/jsx-no-useless-fragment.js

```diff
@@ -0,0 +1,205 @@
+  ecmaVersion: 2018,
+  ecmaFeatures: {
+    jsx: true
+  }
+};
+
+// -----------------------------------------------------------------------------
+// Tests
+// -----------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({parserOptions});
+
+ruleTester.run('jsx-no-uselses-fragment', rule, {
+  valid: [
+    {
+      code: '<><Foo /><Bar /></>',
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<>foo<div /></>',
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<> <div /></>',
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<>{"moo"} </>',
+      parser: parsers.BABEL_ESLINT
+    },
+    '<NotFragment />',
+    '<React.NotFragment />',
+    '<NotReact.Fragment />',
+    {
+      code: '<Foo><><div /><div /></></Foo>',
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<div p={<>{"a"}{"b"}</>} />',
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<Fragment key={item.id}>{item.value}</Fragment>',
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<Fooo content={<>eeee ee eeeeeee eeeeeeee</>} />',
+      parser: parsers.BABEL_ESLINT
+    }
+  ],
+  invalid: [
+    {
+      code: '<></>',
+      output: null,
+      errors: [{messageId: 'NeedsMoreChidren', type: 'JSXFragment'}],
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<p>moo<>foo</></p>',
+      output: '<p>moofoo</p>',
+      errors: [{messageId: 'NeedsMoreChidren'}, {messageId: 'ChildOfHtmlElement'}],
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<>{meow}</>',
+      output: null,
+      errors: [{messageId: 'NeedsMoreChidren'}],
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<p><>{meow}</></p>',
+      output: '<p>{meow}</p>',
+      errors: [{messageId: 'NeedsMoreChidren'}, {messageId: 'ChildOfHtmlElement'}],
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<><div/></>',
+      output: '<div/>',
+      errors: [{messageId: 'NeedsMoreChidren'}],
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: `
+        <>
+          <div/>
+        </>
+      `,
+      output: `
+        <div/>
+      `,
+      errors: [{messageId: 'NeedsMoreChidren'}],
+      parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<Fragment />',
+      errors: [{messageId: 'NeedsMoreChidren'}]
+    },
+    {
+      code: `
+        <React.Fragment>
+          <Foo />
+        </React.Fragment>
+      `,
+      output: `
+        <Foo />
+      `,
+      errors: [{messageId: 'NeedsMoreChidren'}]
+    },
+    {
+      code: `
+        <SomeReact.SomeFragment>
+          {foo}
+        </SomeReact.SomeFragment>
+      `,
+      settings: {
+        react: {
+          pragma: 'SomeReact',
+          fragment: 'SomeFragment'
+        }
+      },
+      errors: [{messageId: 'NeedsMoreChidren'}]
+    },
+    {
+      code: '<Eeee><>foo</></Eeee>',
+      output: '<Eeee>foo</Eeee>',
```

Comment:
actually this one might not be safe, if `Eeee` requires its children be a ReactElement; i think we can only autofix these for html elements, not custom ones.
```suggestion
      output: null,
```

Reply:
indeed